### PR TITLE
Add org.openstream.music

### PR DESCRIPTION
**Pull request by techguy16**
### Please confirm your submission meets all the criteria

- [ x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x ] My pull request follows the instructions at [App Submission][submission].
- [ x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x ] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.openstream.music.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+app-id: org.openstream.music
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '22.08'
```
Comment (bbhtt) [inline org.openstream.music.yaml:6]:
```suggestion
runtime-version: '23.08'
sdk: org.freedesktop.Sdk
base: org.electronjs.Electron2.BaseApp
base-version: '23.08'
```
This is the latest runtime

File: org.openstream.music.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      # Copy the built files to the Flatpak app directory
+      - cp -a ./build/linux-unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ./run.sh
+    subdir: main
+    sources:
+      - type: archive
+        url: https://github.com/openstreamorg/openstreammusic/archive/refs/tags/v1.2.0.tar.gz
+        sha256: 1107d87a71dcf344aab3d84e9e0abbb82f6439b0af8d580f37002365b705d3bb
+      - type: dir
```
Comment (bbhtt) [inline org.openstream.music.yaml:45]: Don't use dir type

File: org.openstream.music.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node18/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: openstreammusic
```
Comment (bbhtt) [inline org.openstream.music.yaml:21]: I don't see appdata, icon or desktop file being installed. All of these should come from the upstream repo.

File: org.openstream.music.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+app-id: org.openstream.music
```
Comment (bbhtt) [inline org.openstream.music.yaml:1]: Do you own or control http://openstream.org/? It redirects to the domain provider

File: org.openstream.music.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.openstream.music.yaml:45]: this will not work anyway. there is no `..`

File: org.openstream.music.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/openstreammusic/flatpak-node/cache
+        npm_config_cache: /run/build/openstreammusic/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      # Change directory to 'desktop'
+      - rm -r index.html
+      - mv desktop/* .
```
Comment (hfiguiere) [inline org.openstream.music.yaml:31]: I don't see why you do that.

File: org.openstream.music.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      # Install npm dependencies
+      - npm install --offline
+      # Build the app using npx electron-builder
+      - npx electron-builder --linux src --x64
```
Comment (hfiguiere) [inline org.openstream.music.yaml:35]: how does this work on aarch64?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I see icon, desktop nor appstream file installed.

File: org.openstream.music.yaml (hunk 4 above)
Comment (hfiguiere) [reply org.openstream.music.yaml:1]: The website from upstream readme is https://openstreamorg.github.io/

So I guess the app-id should be `io.github.openstreamorg.openstreammusic`

**Comment by techguy16** [comment]
I added the icon but can't it get the desktop file from the electron build?

File: org.openstream.music.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      # Copy the built files to the Flatpak app directory
+      - cp -a ./build/linux-unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ./run.sh
+      - install -Dm644 images/logo.png -t /app/share/icons/hicolor/128x128/apps
+      - install -Dm644 images/logo.png -t /app/share/icons/hicolor/48x48/apps
```
Comment (hfiguiere) [inline org.openstream.music.yaml:41]:
```suggestion
```

it surely is not 128x128 and 48x48 at the same time.

**Comment by hfiguiere** [comment]
> I added the icon but can't it get the desktop file from the electron build?

sure it can. However still missing the appstream file.

File: org.openstream.music.yaml (hunk 7 above)
Comment (hfiguiere) [reply org.openstream.music.yaml:41]: btw the icon shall be installed with a name that matches the app-id. And the .desktop have the proper `Icon` field set as well.

**Comment by hfiguiere** [comment]
also even if there is one I don't see it copied in the right place.

**Comment by techguy16** [comment]
So, what can I do to fx it?

File: org.openstream.music.yaml (hunk 4 above)
Comment (hfiguiere) [reply org.openstream.music.yaml:1]: Just to clarify. This is a requirement / blocker.

File: org.openstream.music.yaml (hunk 6 above)
Comment (techguy16) [reply org.openstream.music.yaml:35]: How can I support both?

File: org.openstream.music.yaml (hunk 5 above)
Comment (techguy16) [reply org.openstream.music.yaml:31]: The code to compile is in the desktop folder, as the repository hosts both the web version and the app version.

**Pull request closed without merging**